sdk/cloudmachine/Azure.CloudMachine/src/CoreServices/MessagingServices.cs

```diff
@@ -56,35 +56,37 @@ private ServiceBusSender GetServiceBusSender()
         return sender;
     }
 
-    internal ServiceBusProcessor GetServiceBusProcessor()
+    internal ServiceBusProcessor GetServiceBusProcessor(string id)
     {
         MessagingServices messagingServices = this;
-        ServiceBusProcessor sender = _cm.Subclients.Get(() => messagingServices.CreateProcessor());
+        ServiceBusProcessor sender = _cm.Subclients.Get(() => messagingServices.CreateProcessor(id), id);
         return sender;
     }
 
     private ServiceBusSender CreateSender()
     {
         ServiceBusClient client = GetServiceBusClient();
 
-        ClientConnectionOptions connection = _cm.GetConnectionOptions(typeof(ServiceBusClient));
+        ClientConnectionOptions connection = _cm.GetConnectionOptions(typeof(ServiceBusSender), default);
         ServiceBusSender sender = client.CreateSender(connection.Id);
         return sender;
     }
     private ServiceBusClient CreateClient()
     {
-        ClientConnectionOptions connection = _cm.GetConnectionOptions(typeof(ServiceBusClient));
+        ClientConnectionOptions connection = _cm.GetConnectionOptions(typeof(ServiceBusClient), default);
         ServiceBusClient client = new(connection.Endpoint!.AbsoluteUri, connection.TokenCredential);
         return client;
     }
-    private ServiceBusProcessor CreateProcessor()
+    private ServiceBusProcessor CreateProcessor(string id)
     {
         ServiceBusClient client = GetServiceBusClient();
 
-        ClientConnectionOptions connection = _cm.GetConnectionOptions(typeof(ServiceBusSender));
+        ClientConnectionOptions connection = _cm.GetConnectionOptions(typeof(ServiceBusProcessor), id);
+        string[] topicAndSubscription = connection.Id.Split('/');
+
         ServiceBusProcessor processor = client.CreateProcessor(
-            connection.Id,
-            "cm_servicebus_subscription_private",
+            topicAndSubscription[0],
+            topicAndSubscription[1],
             new() { ReceiveMode = ServiceBusReceiveMode.PeekLock, MaxConcurrentCalls = 5 });
```

Comment:
```suggestion
            new() { MaxConcurrentCalls = 5 });
```

PeekLock is the default.  There's no reason to explicitly specify it.

Reply:
fixed